--- a/drivers/gpu/drm/amd/pm/powerplay/hwmgr/vega20_hwmgr.c
+++ b/drivers/gpu/drm/amd/pm/powerplay/hwmgr/vega20_hwmgr.c
@@ -3816,20 +3816,45 @@
 
 	/* gfxclk */
 	dpm_table = &(data->dpm_table.gfx_table);
+    dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[0].value;
+    dpm_table->dpm_state.soft_max_level = VG20_CLOCK_MAX_DEFAULT;
+    dpm_table->dpm_state.hard_min_level = dpm_table->dpm_levels[0].value;
+    dpm_table->dpm_state.hard_max_level = VG20_CLOCK_MAX_DEFAULT;
+
+    if (PP_CAP(PHM_PlatformCaps_UMDPState)) {
+        if (VEGA20_UMD_PSTATE_GFXCLK_LEVEL < dpm_table->count) {
+            dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[VEGA20_UMD_PSTATE_GFXCLK_LEVEL].value;
+            dpm_table->dpm_state.soft_max_level = dpm_table->dpm_levels[VEGA20_UMD_PSTATE_GFXCLK_LEVEL].value;
+        }
+
+        if (hwmgr->dpm_level == AMD_DPM_FORCED_LEVEL_PROFILE_MIN_SCLK) {
+            dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[0].value;
+            dpm_table->dpm_state.soft_max_level = dpm_table->dpm_levels[0].value;
+        }
+
+        if (hwmgr->dpm_level == AMD_DPM_FORCED_LEVEL_PROFILE_PEAK) {
+            dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[dpm_table->count - 1].value;
+            dpm_table->dpm_state.soft_max_level = dpm_table->dpm_levels[dpm_table->count - 1].value;
+        }
+    }
+
+
+	/* memclk */
+	dpm_table = &(data->dpm_table.mem_table);
 	/* Bypass to prevent UBSAN out-of-bounds read on TB3/ODM Combine requests. */
-    if (dpm_table->count > 0) {
+	if (dpm_table->count > 0) {
         dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[0].value;
         dpm_table->dpm_state.soft_max_level = VG20_CLOCK_MAX_DEFAULT;
         dpm_table->dpm_state.hard_min_level = dpm_table->dpm_levels[0].value;
         dpm_table->dpm_state.hard_max_level = VG20_CLOCK_MAX_DEFAULT;
 
         if (PP_CAP(PHM_PlatformCaps_UMDPState)) {
-            if (VEGA20_UMD_PSTATE_GFXCLK_LEVEL < dpm_table->count) {
-                dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[VEGA20_UMD_PSTATE_GFXCLK_LEVEL].value;
-                dpm_table->dpm_state.soft_max_level = dpm_table->dpm_levels[VEGA20_UMD_PSTATE_GFXCLK_LEVEL].value;
+            if (VEGA20_UMD_PSTATE_MCLK_LEVEL < dpm_table->count) {
+                dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[VEGA20_UMD_PSTATE_MCLK_LEVEL].value;
+                dpm_table->dpm_state.soft_max_level = dpm_table->dpm_levels[VEGA20_UMD_PSTATE_MCLK_LEVEL].value;
             }
 
-            if (hwmgr->dpm_level == AMD_DPM_FORCED_LEVEL_PROFILE_MIN_SCLK) {
+            if (hwmgr->dpm_level == AMD_DPM_FORCED_LEVEL_PROFILE_MIN_MCLK) {
                 dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[0].value;
                 dpm_table->dpm_state.soft_max_level = dpm_table->dpm_levels[0].value;
             }
@@ -3841,30 +3866,6 @@
         }
     }
 
-	/* memclk */
-	dpm_table = &(data->dpm_table.mem_table);
-	dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[0].value;
-	dpm_table->dpm_state.soft_max_level = VG20_CLOCK_MAX_DEFAULT;
-	dpm_table->dpm_state.hard_min_level = dpm_table->dpm_levels[0].value;
-	dpm_table->dpm_state.hard_max_level = VG20_CLOCK_MAX_DEFAULT;
-
-	if (PP_CAP(PHM_PlatformCaps_UMDPState)) {
-		if (VEGA20_UMD_PSTATE_MCLK_LEVEL < dpm_table->count) {
-			dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[VEGA20_UMD_PSTATE_MCLK_LEVEL].value;
-			dpm_table->dpm_state.soft_max_level = dpm_table->dpm_levels[VEGA20_UMD_PSTATE_MCLK_LEVEL].value;
-		}
-
-		if (hwmgr->dpm_level == AMD_DPM_FORCED_LEVEL_PROFILE_MIN_MCLK) {
-			dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[0].value;
-			dpm_table->dpm_state.soft_max_level = dpm_table->dpm_levels[0].value;
-		}
-
-		if (hwmgr->dpm_level == AMD_DPM_FORCED_LEVEL_PROFILE_PEAK) {
-			dpm_table->dpm_state.soft_min_level = dpm_table->dpm_levels[dpm_table->count - 1].value;
-			dpm_table->dpm_state.soft_max_level = dpm_table->dpm_levels[dpm_table->count - 1].value;
-		}
-	}
-
 	/* honour DAL's UCLK Hardmin */
 	if (dpm_table->dpm_state.hard_min_level < (hwmgr->display_config->min_mem_set_clock / 100))
 		dpm_table->dpm_state.hard_min_level = hwmgr->display_config->min_mem_set_clock / 100;
